aiodocker/images.py

```
@@ -0,0 +1,220 @@
+                raise TypeError(
+                    "auth must be base64 encoded string/bytes or a dictionary")
+            if isinstance(auth, bytes):
+                auth = auth.decode('ascii')
+            headers['X-Registry-Auth'] = auth
+        if tag:
+            params['tag'] = tag
+        response = await self.docker._query(
+            "images/{name}/push".format(name=name),
+            "POST",
+            params=params,
+            headers=headers,
+        )
+        return (await json_stream_result(response, stream=stream))
+
+    async def tag(self, name: str, repo: str, tag: Optional[str]=None) -> bool:
+        """
+        Tag the given image so that it becomes part of a repository.
+        Args:
+            repo: the repository to tag in
+            tag: the name for the new tag
+        """
+        params = {"repo": repo}
+
+        if tag:
+            params["tag"] = tag
+
+        await self.docker._query(
+            "images/{name}/tag".format(name=name),
+            "POST",
+            params=params,
+            headers={"content-type": "application/json"},
+        )
+        return True
+
+    async def delete(self, name: str, force: bool=False,
+                     noprune: bool=False) -> List:
+        """
+        Remove an image along with any untagged parent
+        images that were referenced by that image
+        Args:
+            name: name/id of the image to delete
+            force: remove the image even if it is being used
+                   by stopped containers or has other tags
+            noprune: don't delete untagged parent images
+        Returns:
+            List of deleted images
+        """
+        params = {'force': force, 'noprune': noprune}
+        response = await self.docker._query_json(
+            "images/{name}".format(name=name),
+            "DELETE",
+            params=params,
+        )
+        return response
+
+    async def build(self,
```

We should probably raise an error if neither remote or fileobj are specified.